pkg/network/node/egressip.go

```diff
@@ -79,34 +81,49 @@ func (eip *egressIPWatcher) Start(osdnInformers osdninformers.SharedInformerFact
 	return nil
 }
 
-func (eip *egressIPWatcher) Synced() {
+func (eip *egressIPWatcher) SyncIPAssignments() error {
 	link, _, err := GetLinkDetails(eip.localIP)
 	if err != nil {
-		// shouldn't happen, but obviously there's nothing to clean up...
-		return
+		return fmt.Errorf("could not get %s link details %v", eip.localIP, err)
 	}
 	label, err := egressIPLabel(link)
 	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("Could not check for stale egress IPs: %v", err))
-		return
+		return fmt.Errorf("invalid egress IP label: %v", err)
 	}
 	addrs, err := netlink.AddrList(link, netlink.FAMILY_V4)
 	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("Could not check for stale egress IPs: %v", err))
-		return
+		return fmt.Errorf("could not check for egress IPs: %v", err)
 	}
 
+	existingIPs := sets.NewString()
 	for _, addr := range addrs {
 		ip := addr.IP.String()
-		if addr.Label == label && eip.iptablesMark[ip] == "" {
-			klog.Infof("Cleaning up stale egress IP %s", addr.IP.String())
-			err = netlink.AddrDel(link, &addr)
-			if err != nil {
-				utilruntime.HandleError(fmt.Errorf("Could not clean up stale egress IP: %v", err))
+		if addr.Label == label {
+			if eip.iptablesMark[ip] == "" {
+				klog.Infof("Cleaning up stale egress IP %s", addr.IP.String())
+				err = netlink.AddrDel(link, &addr)
+				if err != nil {
+					utilruntime.HandleError(fmt.Errorf("Could not clean up stale egress IP: %v", err))
```

Comment:
we don't like `utilruntime.HandleError` any more anyway (#356), so I'd say just make this use `klog.Error` like the 4.10 version, so there will be fewer conflicts if we need to backport more patches later.

(I think this ended up being the only big difference from the 4.10 version, but if I missed anything else, the same argument may apply there.)